ParselyTracker/Sampler.swift

```diff
@@ -124,6 +128,15 @@ class Sampler {
         }
         self.samplerTimer = Timer.scheduledTimer(withTimeInterval: SAMPLE_RATE, repeats: false) { timer in self.sample() }
     }
+    
+    private func getHeartbeatInterval(existingTimeout: TimeInterval,
+                                      totalTrackedTime: TimeInterval) -> TimeInterval
+    {
+        let totalWithOffset: TimeInterval = totalTrackedTime + offsetMatchingBaseInterval
```

Comment:
New name reads much more easily. A comment explaining the algorithm would still be nice, though having it encapsulated here with an explanatory name implies enough.